### ISSUE TITLE
User request: Methane

### ISSUE THREAD

**Issue by oliver-phet** [phetsims/molecules-and-light#179]
> Friends at PhET, I have a simple request - I discovered the Molecules and Light app last summer when I was writing some lessons related to climate science.
> 
> I love this app for teaching the basics of light/matter interactions, and wonder if there is any chance the model might be expanded to include methane. Doing so would enable us to 
> 
> 1) discuss the role of CO2 as the primary driver of earth's energy balance
> 2) disentangle the "ozone hole" issue from anthropogenic warming and
> 3) discuss methane as a feedback mechanism.
> 
> For us, adding methane would make an already great app greater.

**Comment by jbphet** [comment]
Originally, this simulation was a tab on the Greenhouse Gas simulation, and that tab did include methane.  It wouldn't be hard to add it to the HTML5 version.

![image](https://user-images.githubusercontent.com/3082831/36112963-7e8c89b6-0fe8-11e8-8eb5-db1cb02489d6.png)

**Comment by jessegreenberg** [comment]
@ariel-phet is this something that should be added? @mbarlow12 is going to be working in this sim in the near future for #180, could he work on this issue at the same time?

**Comment by ariel-phet** [comment]
@jessegreenberg this seems like it would be a nice addition considering we are not likely to work on greenhouse anytime soon. Yes it would be reasonable for @mbarlow12 

@arouinfar any thoughts? We should certainly make a mockup (feel free to delegate that to @DianaTavares)

**Comment by arouinfar** [comment]
@ariel-phet I think this would be a great idea!! I talked to several teachers at NSTA who wanted to see The Greenhouse Effect ported to HTML5, so I think this would be a nice addition to MAL.

My primary concern with methane would be how to represent it. It has a tetrahedral geometry, but is presented with a square planar geometry in Java. A quick Unfuddle search reveals that we don't appear to have any complaints, but I still think it would be nice if we could make methane more tetrahedral-like.

I will work with @DianaTavares to mock it up.

**Comment by DianaTavares** [comment]
Thanks! I am interested in participate.

**Comment by arouinfar** [comment]
Great! We can talk about it more when I'm in the office tomorrow @DianaTavares. :smile:

**Comment by DianaTavares** [comment]
this is the sim mockup with the methane molecule 

![screen shot 2018-03-22 at 6 08 48 pm](https://user-images.githubusercontent.com/26031501/37804857-1b3963de-2dfc-11e8-90a0-5626fcf8fc75.png)

Closer mockup of the molecule

![screen shot 2018-03-22 at 6 09 58 pm](https://user-images.githubusercontent.com/26031501/37804886-40ade02c-2dfc-11e8-8809-fbdef8af5b46.png)

and this si the movement of the molecule:
![screen shot 2018-03-22 at 6 10 29 pm](https://user-images.githubusercontent.com/26031501/37804907-6536e402-2dfc-11e8-9c11-542b8ff7dd11.png)

For the molecule movement I used a video that @arouinfar share with me.

**Comment by arouinfar** [comment]
Great, @ariel-phet! 

@mbarlow12 please let me know if you have any questions.

**Comment by mbarlow12** [comment]
I added the basic functionality in the above commit, but I still need to tweak the appearance of the CH4 molecule.

- [ ] TODO: match methane appearance to design

**Comment by samreid** [comment]
In the preceding commit, I fixed a typo that was causing a phet-io fuzzing bug (and probably other issues), please pull.

**Comment by mbarlow12** [comment]
Thanks @samreid for catching.

**Comment by jessegreenberg** [comment]
From #142, the requested 3D appearance of Methane is a bit different from the others and won't be supported by the current representation strategies. I expect it to take ~4 hours to add support for the new appearance and vibration behavior.

**Comment by jessegreenberg** [comment]
In #142 @ariel-phet said

> @jessegreenberg I would like to finish off the methane work.

I discussed with @mbarlow12 and he will work on this soon. Most of the work will go into creating a new Molecule view type that supports layering atoms and atomic bonds, adding support for atomic bonds to not just extend from atom centers, adding support for gradients on the bond nodes, and implementing a vibration strategy to support https://github.com/phetsims/molecules-and-light/issues/179#issuecomment-375498495.

**Comment by mbarlow12** [comment]
New one-off prototype here:
https://phet-dev.colorado.edu/html/molecules-and-light/1.5.0-3dtemp.1/phet/molecules-and-light_en_phet.html

Some pixel polishing necessary still.

**Comment by arouinfar** [comment]
@mbarlow12 the methane vibration is looking really nice! I also kinda like the skewed perspective, as I think it makes it a bit easier to see the vibration. For methane itself, the only pixel polishing I would request would be shorter bonds, similar to the mockup in https://github.com/phetsims/molecules-and-light/issues/179#issuecomment-375801860.

**Comment by zepumph** [comment]
This came up during MAL design meeting. @jessegreenberg @ariel-phet out of curiosity what is the priority on this?

**Comment by jessegreenberg** [comment]
@zepumph and I discussed this today. This is on my list but not at high priority.  It is work to be done after energy-skate-park or when there is a lull for that sim work. RC target for that sim is mid October and I could work on this then.

But please let me know if this is blocking description design/implementation and I could try to work on this sooner.

**Comment by arouinfar** [comment]
@jessegreenberg is there any plan to address the structure of methane before the next publication? In master methane appears to be square planar, but at some point, the molecule had a more tetrahedral structure (see https://github.com/phetsims/molecules-and-light/issues/179#issuecomment-448926228). 

We are likely going to get quite a bit of pushback from users if we use a square planar representation, so I would prefer that we address this before publication.

**Comment by jessegreenberg** [comment]
@arouinfar this is done on master now. The code @mbarlow12 wrote in https://github.com/phetsims/molecules-and-light/commit/834ac2a6a3b9fbd11824a9b8bb7e4dd8b1428e52 was really great, I took the functionality directly from that branch. The only difference is I added the 3D functionality to Molecule instead of creating a new type for methane.

One thing I had trouble with was making space in the radio button group since this molecule is taller than the others. And since all radio buttons have the same height, it made ALL buttons larger. So I scaled down methane in particular. @arouinfar does this look OK?
